csharp/src/Ice/LocatorInfo.cs

```diff
@@ -229,6 +238,75 @@ internal void ClearCache(Reference reference)
             return (endpoints, cached);
         }
 
+        // Encoding represents the encoding used to send the getRegistry request.
+        internal async ValueTask<ILocatorRegistryPrx?> GetLocatorRegistryAsync(Encoding encoding)
+        {
+            if (_locatorRegistry == null)
+            {
+                ILocatorRegistryPrx? prx =
+                    await Locator.Clone(encoding: encoding).GetRegistryAsync().ConfigureAwait(false);
+
+                // The locator registry can't be located and we use ordered endpoint selection in case the locator
+                // returned a proxy with some endpoints which are preferred to be tried first.
+                _locatorRegistry = prx?.Clone(clearLocator: true, endpointSelection: EndpointSelectionType.Ordered);
+            }
+            return _locatorRegistry;
+        }
+
+        // Returns true if the time-to-live has been reached
+        private static bool CheckTTL(TimeSpan insertionTime, TimeSpan ttl) =>
+            ttl == Timeout.InfiniteTimeSpan || (Time.Elapsed - insertionTime) <= ttl;
+
+        private static void Trace(string msg, Reference r, IReadOnlyList<Endpoint> endpoints)
+        {
+            var s = new System.Text.StringBuilder();
```

Comment:
we can add `using System.Text` and simplify this and other usages in this file, also we typically use `sb` for the builder

Reply:
Fixed.